src/player_activity.cpp

```diff
@@ -549,100 +191,50 @@ bool containers_equal( const T &left, const T &right )
-        // that easy resuming isn't possible
-        case ACT_FORAGE:
-        case ACT_OPEN_GATE:
-        case ACT_OXYTORCH:
-        case ACT_CRACKING:
-        case ACT_FISH:
-        case ACT_PICKAXE:
-        case ACT_BURROW:
-        // Those should check position
-        // But position isn't set here yet!
-        // @todo Update the functions that set those activities
-        case ACT_BUTCHER:
-        case ACT_TRAIN:
-        case ACT_HOTWIRE_CAR:
-        case ACT_START_ENGINES:
-        case ACT_BUILD:
-        case ACT_VEHICLE:
-        case ACT_FIRSTAID:
-        case ACT_VIBE:
-        case ACT_MAKE_ZLAVE:
-        case ACT_GUNMOD_ADD:
-        case ACT_REPAIR_ITEM:
-        case ACT_MEND_ITEM:
-        case ACT_MEDITATE:
-            // Those should have extra limitations
-            // But for now it's better to allow too much than too little
-            break;
-        case ACT_READ:
-            // Return false if any NPCs joined or left the study session
-            // the vector {1, 2} != {2, 1}, so we'll have to check manually
-            if( values.size() != other.values.size() ) {
-                return false;
-            }
-            for( int foo : other.values ) {
-                if( std::find( values.begin(), values.end(), foo ) == values.end() ) {
-                    return false;
-                }
-            }
-            if( targets.empty() || other.targets.empty() || targets[0] != other.targets[0] ) {
+        }
+    } else if( id() == activity_id( "ACT_READ" ) ) {
+        // Return false if any NPCs joined or left the study session
+        // the vector {1, 2} != {2, 1}, so we'll have to check manually
+        if( values.size() != other.values.size() ) {
+            return false;
+        }
+        for( int foo : other.values ) {
+            if( std::find( values.begin(), values.end(), foo ) == values.end() ) {
                 return false;
             }
-            break;
-
-        case ACT_CLEAR_RUBBLE:
-            if( other.coords.empty() || other.coords[0] != coords[0] ) {
+        }
+        if( targets.empty() || other.targets.empty() || targets[0] != other.targets[0] ) {
+            return false;
+        }
+    } else if( id() == activity_id( "ACT_WAIT" ) || id() == activity_id( "ACT_WAIT_NPC" ) ||
+               id() == activity_id( "ACT_WAIT_WEATHER" ) || id() == activity_id( " ACT_ARMOR_LAYERS" ) ||
+               id() == activity_id( "ACT_PULP" ) || id() == activity_id( "ACT_AIM" ) ||
+               id() == activity_id( "ACT_LONGSALVAGE" ) || id() == activity_id( "ACT_ATM" ) ||
+               id() == activity_id( "ACT_DROP" ) || id() == activity_id( "ACT_STASH" ) ||
+               id() == activity_id( "ACT_PICKUP" ) || id() == activity_id( " ACT_MOVE_ITEMS" ) ||
+               id() == activity_id( "ACT_ADV_INVENTORY" ) || id() == activity_id( " ACT_START_FIRE" ) ||
+               id() == activity_id( "ACT_FILL_LIQUID" ) ) {
```

Comment:
That's a large and ugly block, would be better to make it a property, like `no_resume`.

Also, some of the entries here have extra spaces in the id.